Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/CruiseControl.cs

```diff
@@ -0,0 +1,2277 @@
+                SelectedMaxAccelerationStep == 0 && DisableCruiseControlOnThrottleAndZeroForce && ForceRegulatorAutoWhenNonZeroSpeedSelectedAndThrottleAtZero &&
+                            Locomotive.ThrottleController.CurrentValue == 0 && Locomotive.DynamicBrakeController.CurrentValue == 0))
+            {
+                SpeedRegMode = SpeedRegulatorMode.Auto;
+            }
+            if (SpeedRegMode != SpeedRegulatorMode.Auto && (ForceRegulatorAutoWhenNonZeroSpeedSelectedAndThrottleAtZero &&
+                SelectedMaxAccelerationStep == 0 && DisableCruiseControlOnThrottleAndZeroForceAndZeroSpeed &&
+                            Locomotive.ThrottleController.CurrentValue == 0 && Locomotive.DynamicBrakeController.CurrentValue == 0))
+            {
+                SpeedRegMode = SpeedRegulatorMode.Auto;
+            }
+            if (UseThrottleAsSpeedSelector || HasProportionalSpeedSelector)
+            {
+                selectedSpeedIncreasing = true;
+                if (SelectedSpeedMpS == 0)
+                {
+                    Locomotive.SignalEvent(Common.Event.LeverFromZero);
+                }
+            }
+            else
+                SpeedSelectorModeStartIncrease();
+        }
+        public void SpeedRegulatorSelectedSpeedStopIncrease()
+        {
+            if (Locomotive.MultiPositionControllers != null)
+            {
+                foreach (Controllers.MultiPositionController mpc in Locomotive.MultiPositionControllers)
+                {
+                    if (mpc.controllerBinding != Controllers.MultiPositionController.ControllerBinding.SelectedSpeed)
+                        return;
+                    mpc.StateChanged = false;
+                    mpc.DoMovement(Controllers.MultiPositionController.Movement.Neutral);
+                    return;
+                }
+            }
+            if (UseThrottleAsSpeedSelector || HasProportionalSpeedSelector)
+                selectedSpeedIncreasing = false;
+            else
+                SpeedSelectorModeStopIncrease();
+        }
+
+        protected double selectedSpeedLeverHoldTime = 0;
+        public void SpeedRegulatorSelectedSpeedIncrease()
+        {
+            if (!Equipped) return;
+
+            if (selectedSpeedLeverHoldTime + SpeedSelectorStepTimeSeconds > TotalTime)
+                return;
+            selectedSpeedLeverHoldTime = TotalTime;
+
+            SelectedSpeedMpS = Math.Max(MinimumSpeedForCCEffectMpS, SelectedSpeedMpS + SpeedRegulatorNominalSpeedStepMpS);
+            if (SelectedSpeedMpS > Locomotive.MaxSpeedMpS)
+                SelectedSpeedMpS = Locomotive.MaxSpeedMpS;
+            if (SpeedIsMph)
+                Simulator.Confirmer.Message(ConfirmLevel.Information, Simulator.Catalog.GetString("Selected speed changed to ") + Math.Round(MpS.FromMpS(SelectedSpeedMpS, false), 0, MidpointRounding.AwayFromZero).ToString() + " mph");
```

Comment:
Localization

Reply:
What do you mean here? Localizing "mph"?

Reply:
I meant the same as in other lines, that you use "speed changed to {0} km/h" instead of "speed changed to "+speed+" km/h". I don't think it's necessary to localize "mph" nor "kph". You should include the units into the localization string, though, instead of assuming that they always come at the end of the string. I don't know if that is the case, but there might be a language where the syntax is Yoda-like "to {0} km/h speed changed".

Reply:
Done